node/basic-cloud-functions-bot/index.js

```diff
@@ -18,190 +18,196 @@
+  switch (cleanText(body.text)) {
+    case 'i love dogs':
       return getInteractiveCard();
       break;
   }
 }
 
 /**
- * Handles the interaction when a card is clicked. Returns a single image
+ * Handles the interaction when a card is clicked. Returns a single image.
  * @param {Object} body
- * @returns {Object} returns a card object with the proper actionResponse to update the existing comment
+ * @return {Object} returns a card object with the proper actionResponse to
+ *     update the existing comment
  */
 function handleCardClick(body) {
-  let { cards } = getSingleImage(body.action.parameters[0].value);
+  let {cards} = getSingleImage(body.action.parameters[0].value);
   return {
     actionResponse: {
-      type: "UPDATE_MESSAGE"
+      type: 'UPDATE_MESSAGE',
     },
-    cards
+    cards,
   };
 }
 
 /**
- * Removed bot name, extra spaces, and returns back the important part of the text sent
+ * Removed bot name, extra spaces, and returns back the important part of the
+ * text sent.
  * @param {String} text
  * @return {String}
  */
 function cleanText(text) {
   return text
     .toLowerCase()
-    .replace(`@${BOTNAME.toLowerCase()}`, "")
-    .split(" ")
-    .filter(char => char !== "")
-    .join(" ");
+    .replace(`@${BOT_NAME.toLowerCase()}`, '')
+    .split(' ')
+    .filter((char) => char !== '')
+    .join(' ');
 }
 
 /**
- * Returns a static interactive card, you can map over arrays to populate something similar
- * @returns {Object}
+ * Returns a static interactive card, you can map over arrays to populate
+ * something similar.
+ * @return {Object}
  */
 function getInteractiveCard() {
   let imageUrls = [
-    "https://photos.app.goo.gl/5noEuKjptkjWd01n1",
-    "https://photos.app.goo.gl/DRqnKP4YBO12SnLT2"
+    'https://photos.app.goo.gl/5noEuKjptkjWd01n1',
+    'https://photos.app.goo.gl/DRqnKP4YBO12SnLT2',
   ];
   return {
     cards: [{
         header: {
-          title: "Which type",
-          subtitle: "Click on to Choose",
+          title: 'Which type',
+          subtitle: 'Click on to Choose',
           imageUrl:
-            "http://www.freepngimg.com/download/dog/9-dog-png-image-picture-download-dogs.png",
-          imageStyle: "IMAGE"
+            'http://www.freepngimg.com/download/dog/9-dog-png-image-picture-download-dogs.png',
+          imageStyle: 'IMAGE',
         },
         sections: [{
-            widgets: getWidgets(imageUrls, getImageButtonWidget)
-          }]
-      }]
+            widgets: getWidgets(imageUrls, getImageButtonWidget),
+          }],
+      }],
   };
 }
 
 /**
- * Return an array of widgets
+ * Return an array of widgets.
  * @param {Array} urls a simple array of imageUrls, strings
  * @param {Function} widgetComposer a function used to compose a widget object
- * @returns {Array}
+ * @return {Array}
  */
 function getWidgets(urls, widgetComposer) {
-  return urls.map(url => widgetComposer(url));
+  return urls.map((url) => widgetComposer(url));
```

Comment:
This is unnecessarily unboxing `url` and re-wrapping it.

```js
return urls.map(widgetComposer);
```

You can probably inline this method.

Reply:
Done